### ISSUE TITLE
the request.resource isnot working to angularfire v9

### ISSUE THREAD

**Issue by AdamTeodoro** [angular/angularfire#3054]
The rule that use request.resource.data is not working to angularfire V9

**Comment by AdamTeodoro** [comment]
they wouldn't solve it even if I had followed the rules for publishing problems kkkkkk, but at least that way I complain and don't waste time typing too much

**Comment by jamesdaniels** [comment]
@AdamTeodoro are you talking about security rules?

**Comment by jamesdaniels** [comment]
I understand you're frustrated but please be mindful of the [Code of Conduct](https://github.com/angular/code-of-conduct/blob/master/CODE_OF_CONDUCT.md). We're volunteers here & we have competing demands for our attention.

**Comment by AdamTeodoro** [comment]
This is not frustrating for me, i 'm sorry the problem is very visible i'm commit for see if devs are update about this problem, without more explications.

Is this one básic use:
[look my project ](https://github.com/AdamTeodoro/AdamTeodoro)

the timestamp is not working:

This rule is notworking with angularfirev9:
my angular code:
  my component:
  https://github.com/AdamTeodoro/AdamTeodoro/tree/master/src/app/components/commit
  my angualar service:
  https://github.com/AdamTeodoro/AdamTeodoro/blob/master/src/app/services/commit.service.ts
my rule
request.time > resource.data.timestamp + duration.value(2, 's');
---------------------------------------------------------------------------------------------
my angular code: 
https://github.com/AdamTeodoro/AdamTeodoro/blob/master/src/app/app.component.ts
my rule:
resource.data.accesses + 1 == request.resource.data.accesses;

**Comment by AdamTeodoro** [comment]
i'm sorry for my inglish, and thanks for your service and for your patience.

**Comment by jamesdaniels** [comment]
I'm not sure what you're trying to achieve with that security rule, that 2 seconds must have elapsed from the time that the data was saved? Since you're using server time it strikes me that would be expected to fail.

**Comment by AdamTeodoro** [comment]
yes, the first rule is for limit the access per second.
the second rule is for limit the increment number for accesse.

**Comment by AdamTeodoro** [comment]
yes, the first rule is to limit access, forcing the user to wait seconds to perform a new access.
The second rule is to limit the number of increments per access, for each access to increment only once.
These rules are working in the previous version.

**Comment by AdamTeodoro** [comment]
Me desculpe pela demora durante esses quatro dias eu estava dormindo.

**Comment by AdamTeodoro** [comment]
I'm reading other cases and the problem could be that I'm using set and not update.
